vector/v.info/print.c

```diff
@@ -77,50 +94,81 @@ void print_topo(struct Map_info *Map)
+                Vect_get_num_primitives(Map, GV_CENTROID));
         fflush(stdout);
 
-        fprintf(stdout, "kernels=%d\n",
-                Vect_get_num_primitives(Map, GV_KERNEL));
+        fprintf(stdout, "areas=%d\n", Vect_get_num_areas(Map));
         fflush(stdout);
 
-        fprintf(stdout, "volumes=%d\n",
-                Vect_get_num_primitives(Map, GV_VOLUME));
+        fprintf(stdout, "islands=%d\n", Vect_get_num_islands(Map));
         fflush(stdout);
 
-        fprintf(stdout, "holes=%d\n", Vect_get_num_holes(Map));
+        if (with_z) {
+            fprintf(stdout, "faces=%d\n",
+                    Vect_get_num_primitives(Map, GV_FACE));
+            fflush(stdout);
+
+            fprintf(stdout, "kernels=%d\n",
+                    Vect_get_num_primitives(Map, GV_KERNEL));
+            fflush(stdout);
+
+            fprintf(stdout, "volumes=%d\n",
+                    Vect_get_num_primitives(Map, GV_VOLUME));
+            fflush(stdout);
+
+            fprintf(stdout, "holes=%d\n", Vect_get_num_holes(Map));
+            fflush(stdout);
+        }
+
+        fprintf(stdout, "primitives=%ld\n", nprimitives);
         fflush(stdout);
-    }
 
-    fprintf(stdout, "primitives=%ld\n", nprimitives);
-    fflush(stdout);
+        fprintf(stdout, "map3d=%d\n", Vect_is_3d(Map) ? 1 : 0);
+        fflush(stdout);
 
-    fprintf(stdout, "map3d=%d\n", Vect_is_3d(Map) ? 1 : 0);
-    fflush(stdout);
+        break;
+    case JSON:
+        json_object_set_number(root_object, "nodes", Vect_get_num_nodes(Map));
+        json_object_set_number(root_object, "points",
+                               Vect_get_num_primitives(Map, GV_POINT));
+        json_object_set_number(root_object, "lines",
+                               Vect_get_num_primitives(Map, GV_LINE));
+        json_object_set_number(root_object, "boundaries",
+                               Vect_get_num_primitives(Map, GV_BOUNDARY));
+        json_object_set_number(root_object, "centroids",
+                               Vect_get_num_primitives(Map, GV_CENTROID));
+        json_object_set_number(root_object, "areas", Vect_get_num_areas(Map));
+        json_object_set_number(root_object, "islands",
+                               Vect_get_num_islands(Map));
+        if (with_z) {
+            json_object_set_number(root_object, "faces",
+                                   Vect_get_num_primitives(Map, GV_FACE));
+            json_object_set_number(root_object, "kernels",
+                                   Vect_get_num_primitives(Map, GV_KERNEL));
+            json_object_set_number(root_object, "volumes",
+                                   Vect_get_num_primitives(Map, GV_VOLUME));
+            json_object_set_number(root_object, "holes",
+                                   Vect_get_num_holes(Map));
+        }
+        json_object_set_number(root_object, "primitives", nprimitives);
+        json_object_set_number(root_object, "map3d", Vect_is_3d(Map) ? 1 : 0);
```

Comment:
While technically, [Vect_is_3d](https://grass.osgeo.org/programming8/defs_2vector_8h.html#af093480c5eef1ca4c0a24ab95718bce2) returns

```C
 #define WITHOUT_Z              0
 #define WITH_Z                 1
```

it is meant to be interpreted as a boolean according the doc. Hence,

```suggestion
        json_object_set_boolean(root_object, "map3d", Vect_is_3d(Map));
```

The shell-style output is designed with the idea that there is no special boolean type available, but JSON has that type.